Build issues against latest git

I'm afraid we're here again, against latest git:
```
FAILED: src/libkbdd.so.p/kbdd.cpp.o 
ccache c++ -Isrc/libkbdd.so.p -Isrc -I../src -I/usr/include/cairo -I/usr/include/lzo -I/usrnclude/libpng16 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -Isr/lib/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/pixman-1 -I/usr/include/pan-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/libm -I/usr/include/libevdev-1.0 -I/usr/include/libxml2 -fdiagnostics-color=always -D_FILE_OFFT_BITS=64 -Wall -Winvalid-pch -Wextra -std=c++17 -O0 -g -DWLR_USE_UNSTABLE -DWAYFIRE_PLUGINfPIC -pthread -MD -MQ src/libkbdd.so.p/kbdd.cpp.o -MF src/libkbdd.so.p/kbdd.cpp.o.d -o src/bkbdd.so.p/kbdd.cpp.o -c ../src/kbdd.cpp
../src/kbdd.cpp:63:9: error: ‘signal_connection_t’ in namespace ‘wf’ does not name a type
   63 |     wf::signal_connection_t keyboard_focus_changed = [=] (wf::signal_data_t *data) 
      |         ^~~~~~~~~~~~~~~~~~~
../src/kbdd.cpp: In member function ‘virtual void kbdd_plugin::init()’:
../src/kbdd.cpp:88:24: error: ‘class wf::compositor_core_t’ has no member named ‘connect_sial’
   88 |         wf::get_core().connect_signal("keyboard-focus-changed", &keyboard_focus_chaed);
      |                        ^~~~~~~~~~~~~~
../src/kbdd.cpp:88:66: error: ‘keyboard_focus_changed’ was not declared in this scope
   88 |         wf::get_core().connect_signal("keyboard-focus-changed", &keyboard_focus_chaed);
      |                                                                  ^~~~~~~~~~~~~~~~~~~~
../src/kbdd.cpp: In member function ‘virtual void kbdd_plugin::fini()’:
../src/kbdd.cpp:92:24: error: ‘class wf::compositor_core_t’ has no member named ‘disconnectignal’
   92 |         wf::get_core().disconnect_signal(&keyboard_focus_changed);
      |                        ^~~~~~~~~~~~~~~~~
../src/kbdd.cpp:92:43: error: ‘keyboard_focus_changed’ was not declared in this scope
   92 |         wf::get_core().disconnect_signal(&keyboard_focus_changed);

```

_Originally posted by @stefonarch in https://github.com/AlexJakeGreen/wayfire-kbdd-plugin/issues/6#issuecomment-1401528303_

I was still on the 0.8 branch I see, in master the "typos" are gone but basically it's the same:
```
./src/kbdd.cpp:63:9: error: ‘signal_connection_t’ in namespace ‘wf’ does not name a type
   63 |     wf::signal_connection_t keyboard_focus_changed = [=] (wf::signal_data_t *data) {
      |         ^~~~~~~~~~~~~~~~~~~
../src/kbdd.cpp: In member function ‘virtual void kbdd_plugin::init()’:
../src/kbdd.cpp:88:24: error: ‘class wf::compositor_core_t’ has no member named ‘connect_signal’
   88 |         wf::get_core().connect_signal("keyboard-focus-changed", &keyboard_focus_changed);
      |                        ^~~~~~~~~~~~~~
../src/kbdd.cpp:88:66: error: ‘keyboard_focus_changed’ was not declared in this scope
   88 |         wf::get_core().connect_signal("keyboard-focus-changed", &keyboard_focus_changed);
      |                                                                  ^~~~~~~~~~~~~~~~~~~~~~
../src/kbdd.cpp: In member function ‘virtual void kbdd_plugin::fini()’:
../src/kbdd.cpp:92:24: error: ‘class wf::compositor_core_t’ has no member named ‘disconnect_signal’
   92 |         wf::get_core().disconnect_signal(&keyboard_focus_changed);
      |                        ^~~~~~~~~~~~~~~~~
../src/kbdd.cpp:92:43: error: ‘keyboard_focus_changed’ was not declared in this scope
   92 |         wf::get_core().disconnect_signal(&keyboard_focus_changed);
      |                                           ^~~~~~~~~~~~~~~~~~~~~~c
ninja: build stopped: subcommand failed.

```

--- a/src/kbdd.cpp
+++ b/src/kbdd.cpp
@@ -60,8 +60,7 @@ class kbdd_plugin : public wf::plugin_interface_t
     };
 
 
-    wf::signal_connection_t keyboard_focus_changed = [=] (wf::signal_data_t *data) {
-        wf::keyboard_focus_changed_signal *signal = static_cast<wf::keyboard_focus_changed_signal*>(data);
+    wf::signal::connection_t<wf::keyboard_focus_changed_signal> keyboard_focus_changed = [=] (wf::keyboard_focus_changed_signal *signal) {
         wf::scene::node_ptr focused_node = signal->new_focus;
         wayfire_view view = wf::node_to_view(focused_node);
 
@@ -85,11 +84,11 @@ class kbdd_plugin : public wf::plugin_interface_t
 public:
     void init() override
     {
-        wf::get_core().connect_signal("keyboard-focus-changed", &keyboard_focus_changed);
+        wf::get_core().connect(&keyboard_focus_changed);
     }
 
     void fini() override {
-        wf::get_core().disconnect_signal(&keyboard_focus_changed);
+        wf::get_core().disconnect(&keyboard_focus_changed);
     }
 };
 

